### ISSUE TITLE
attributes warns if attribute used on the line above for struct

### ISSUE THREAD

**Issue by bobbradley** [realm/SwiftLint#4884]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If you opt into the "attributes" warning, it warns if you use an attribute on a line above for a type. The documentation page (https://realm.github.io/SwiftLint/attributes.html) indicates it should not warn if it's used on the same line for imports (which works) or on a separate line for type (that part warns for me).

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint TestAttributes.swift
Linting Swift files at paths TestAttributes.swift
Linting 'TestAttributes.swift' (1/1)
TestAttributes.swift:5:8: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)
Done linting! Found 1 violation, 0 serious in 1 file
```

### Environment

* SwiftLint version: 0.51.0
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - attributes
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Not using with Xcode.
* Do you have a sample that shows the issue? Yes, below:

```swift
@_spi(Private) import SomeFramework

/// Test documentation.
@_spi(Private)
public struct MyStruct {
}
```

**Comment by SimplyDanny** [comment]
Are you sure the rule reports a violation on the struct definition and not on the import statement? I see the warning on the import statement only which is expected starting from 0.51.0 since the attribute has an argument. There is #4855 in the works to disable this behavior, however.

**Comment by bobbradley** [comment]
You're right, when I have an argument, it warns about the import. But if I don't have an argument then I see it warn on the type (which was the original problem I was seeing). Would https://github.com/realm/SwiftLint/pull/4855 also fix that part so I could have the following without any warnings?

```
@_spi(Private) import SomeFramework

/// Test documentation.
@_spi(Private)
public struct MyStruct {
}
```

**Comment by SimplyDanny** [comment]
> Would #4855 also fix that part so I could have the following without any warnings?
> 
> ```
> @_spi(Private) import SomeFramework
> 
> /// Test documentation.
> @_spi(Private)
> public struct MyStruct {
> }
> ```

Yes. However, `@_spi` without arguments is a compilation error. SwiftLint requires the input to be compilable Swift code. As `@_spi` is not valid, SwiftLint does something strange. With `@_spi(Something)` all is fine.

So all that remains for the issue with the import statement is #4855 to be merged and released. The issue with `@_spi` you can fix yourself. 😉

**Comment by bobbradley** [comment]
Sounds good. Duplicate of https://github.com/realm/SwiftLint/pull/4855.
